### PR TITLE
docs: Set up documentation with MkDocs (DEV-4134)

### PR DESCRIPTION
**Pull request by BalduinLandolt**
This PR provides setup for publishing the documentation to docs.dasch.swiss by setting up MkDocs.  
It does not yet contain significant changes to contents of the documentation; this will happen in a later PR.

### CODE REVIEW

File: .markdownlint.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+
+# Config file for https://github.com/igorshubovych/markdownlint-cli
```
Comment (mpro7) [inline .markdownlint.yml:3]:
```suggestion
# Config file for https://github.com/igorshubovych/markdownlint-cli
```
Is this something to keep or rather some mistaken keyboard hits?

File: docs/adding-metadata.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Adding Metadata
+
+To add metadata to the DSP-META repository,
```
Comment (mpro7) [inline docs/adding-metadata.md:3]:
```suggestion
# Adding Metadata

To add metadata to the [DSP-META repository](https://github.com/dasch-swiss/dsp-meta),
```

**Review by mpro7 — APPROVED** [review]
Looks fine, just check my suggestions below:

File: docs/data/api.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# API Documentation
+
+The pubic API of the DSP Metadata Browser can be accessed under `https://meta.dasch.swiss/api`.
+
+## Endpoints
+
+Get all projects: `GET /v1/projects`
```
Comment (seakayone) [inline docs/data/api.md:7]:
```suggestion
Get all projects with paging: `GET /v1/projects`
Query params `_limit` (default value `100`) and `_page` (default value `1`) allow you to define the page.

```

**Review by seakayone — APPROVED** [review]
Apart from the missing paging this looks good to me.

File: .markdownlint.yml (hunk 1 above)
Comment (BalduinLandolt) [reply .markdownlint.yml:3]: I copied this over from the API repo, but you're right, probably not relevant.

**Pull request merged**